src/sentry/migrations/0687_alert_rule_project_backfill_migration.py

```diff
@@ -25,7 +25,28 @@ def _backfill_alert_rule_projects(apps, schema_editor):
             )
             continue
 
-        alert_rule = snuba_query.alertrule_set.get()
+        alert_rule_set = list(snuba_query.alertrule_set.all())
+        if not len(alert_rule_set):
+            logger.warning(
+                "QuerySubscription + SnubaQuery found with no alert_rule",
+                extra={
+                    "query_subscription_id": subscription.id,
+                    "snuba_query_id": snuba_query.id,
+                },
+            )
+            continue
+        elif len(alert_rule_set) > 1:
```

Comment:
Even though i'm handling this case here

https://redash.getsentry.net/queries/6449/source

can confirm there are no cases where there is more than 1 alert rule pointing to a single snuba_query 🙏